server/channels/store/sqlstore/user_store.go

```diff
@@ -184,10 +184,6 @@ func (us SqlUserStore) Update(user *model.User, trustedUpdateData bool) (*model.
 	}
 
 	user.CreateAt = oldUser.CreateAt
-	user.AuthData = oldUser.AuthData
-	user.AuthService = oldUser.AuthService
-	user.Password = oldUser.Password
-	user.LastPasswordUpdate = oldUser.LastPasswordUpdate
```

Comment:
I'm worried it's going to be hard to evaluate the system-wide consequences of this change. Instead of touching this, could we just expose `SqlUserStore.UpdateAuthData` through the `app` and plugin API as a discrete method?